bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/DeviceHandler.java

```diff
@@ -0,0 +1,313 @@
+    }
+
+    @Override
+    public void apiConnectionStatusChanged(ConnectionStatus connectionStatus) {
+        if (ConnectionStatus.SUCCESS.equals(connectionStatus)) {
+            ThingStatus status = getThing().getStatus();
+            if (status != ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+                // Wait a little bit before refreshing because a dispose maybe running in parallel
+                scheduler.schedule(() -> scheduleRefreshJob(), 3, TimeUnit.SECONDS);
+            }
+        } else {
+            freeRefreshJob();
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, connectionStatus.getMessage());
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Running dispose()");
+        refreshStrategy = null;
+        freeRefreshJob();
+        apiBridge.removeConnectionListener(this);
+        getBridgeHandler().ifPresent(handler -> handler.unregisterDataListener(this));
+    }
+
+    private void freeRefreshJob() {
+        ScheduledFuture<?> job = refreshJob;
+        if (job != null) {
+            job.cancel(true);
+        }
+        refreshJob = null;
+    }
+
+    private void scheduleRefreshJob() {
+        RefreshStrategy strategy = refreshStrategy;
+        if (strategy != null) {
+            long delay = strategy.nextRunDelay().toSeconds();
+            logger.debug("Scheduling update channel thread in {} s", delay);
+            updateChannels(false);
+            freeRefreshJob();
+            refreshJob = scheduler.schedule(() -> scheduleRefreshJob(), delay, TimeUnit.SECONDS);
+        }
+    }
+
+    protected @Nullable NAThing updateReadings() throws NetatmoException {
+        throw new NetatmoException("Should not be called");
+    }
+
+    private synchronized void updateChannels(boolean requireDefinedRefreshInterval) {
+        RefreshStrategy strategy = refreshStrategy;
+        if (strategy != null) {
+            boolean dataOutdated = (requireDefinedRefreshInterval && strategy.isSearchingRefreshInterval()) ? false
+                    : strategy.isDataOutdated();
+            if (dataOutdated) {
+                logger.debug("Trying to update channels on device {}", config.id);
+                try {
+                    NAThing newData = updateReadings();
+                    if (newData != null) {
+                        logger.debug("Successfully updated device {} readings! Now updating channels", config.id);
+                        updateProperties(newData);
+                        setNewData(newData);
+                        newData.getLastSeen().ifPresent(ts -> strategy.setDataTimeStamp(ts));
+                        updateChildModules(newData);
+                    }
+                } catch (NetatmoException e) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Unable to connect Netatmo API : " + e.getLocalizedMessage());
```

Comment:
Provide i18n text?

Reply:
I have not yet look at this part of the code but I think you should very probably extend my new core class in case you would like to implement internationalization of exceptions.

Reply:
I'm referring only to localization of thing status text. But I can see there is also `getLocalizedMessage` from exception which is probably what you are commenting?

Reply:
Yes, you're right, that is 2 different things ;)
With the current implementation of NetatmoException, using e.getLocalizedMessage() has no sense.